src/lfn.c

```diff
@@ -69,6 +68,14 @@ WFFindFirst(
 
    lpFind->fd.dwFileAttributes &= ATTR_USED;
 
+   if (lpFind->fd.dwFileAttributes & FILE_ATTRIBUTE_REPARSE_POINT) {
+       if (lpFind->fd.dwReserved0 == IO_REPARSE_TAG_MOUNT_POINT) {
+           lpFind->fd.dwFileAttributes |= ATTR_JUNCTION;
+       } else if (lpFind->fd.dwReserved0 == IO_REPARSE_TAG_SYMLINK) {
+           lpFind->fd.dwFileAttributes |= ATTR_SYMBOLIC;
+       }
+   }
+

```

Comment:
This code is a much better way to do what DecodeReparsePoint does.  We should be able to remove one or both calls to DecodeReparsePoint.

Reply:
You might want to omit evaluating attributes returned from DecodeReparsePoint(), but you can't remove the call to it, because it delivers szLinkDest, which is needed.

Reply:
I changed the one caller that can be changed.